services/web/server/src/simcore_service_webserver/projects/projects_handlers.py

```diff
@@ -56,11 +56,19 @@ async def open_project(request: web.Request) -> web.Response:
             include_state=True,
         )
 
+        repo = ProductRepository(request)
+        max_number_of_studies_per_user = (
+            await repo.get_product_max_number_studies_open_per_user(
```

Comment:
This solution would always check in the database. 

An alternative is to check in the product instance generated upon start. The advantage is that it does not access the database. The disadvantage is that if you change the database, it refreshes only upon start. 

```python
from .products import get_current_product, Product
# ...

product: Product = get_current_product(request)

# ...
... (max_number_of_studies_per_user = product.max_open_studies_per_user)
```

```

Reply:
that is very very nice and elegant! will definitely use this.